pulser/simulation/simulation.py

```diff
@@ -241,7 +265,29 @@ def make_vdw_term() -> float:
                 vdw += U * self._build_operator('sigma_rr', q1, q2)
             return vdw
 
-        def build_coeffs_ops(basis: str, addr: str) -> list[list]:
+        def make_interaction_xy_term():
+            """Construct the XY interaction Term.
+
+            For each pair of qubits, calculate the distance between them,
+            then assign the local operator "sigma_du * sigma_ud" at each pair.
+            The units are given so that the coefficient
+            includes a 1/hbar factor.
+            """
+            xy = 0
+            # Get every pair without duplicates
+            for q1, q2 in itertools.combinations(self._qdict.keys(), r=2):
+                dist = np.linalg.norm(
+                    self._qdict[q1] - self._qdict[q2])
+                cosine = 0.
```

Comment:
A comment for future implementation: The cosine would be taken with the angle between the magnetic field axis and the atoms' plane. This is to be specified in the device.

Reply:
What about the possibility to change the magnetic field between experiments ?

Reply:
At first I thought that adjusting this was a very signfiicant change (such as, e.g. changing the principal quantum number of the rydberg level(s) ), but if this complexity is not as relevant, the next place where the angle would be read is as an (optional) argument in `Sequence.declare_channel()`. For example:

```python
my_seq.declare_channel('ch_xy', 'mw_global', axis_angle=np.pi/2)
```

Reply:
At first I thought that adjusting this was a very signficant change (such as, e.g. changing the principal quantum number of the rydberg level(s) ), but if this complexity is not as relevant, the next place where the angle would be read is as an (optional) argument in `Register`. For example:

```python
my_reg._axis_angle = np.pi/2
```

I don't like the option of including this information in `Sequence`(Since this holds only information about the channels implementing the Rabi oscillations, and not the applied magnetic field defining the quantization axis). Neither in `Simulation`, since the magnetic field is a structural component of the XY setup.

As long as there's no structure for changing the `Device` specs, let's try to put this into `Register` (What do you think, @HGSilveri , @lhenriet ?).

Reply:
Shouldn't it be a vector that you specify ?

Reply:
Ah, yes, I was thinking in 1D. In polar coordinates it would be something like `my_reg._axis_angle = [theta, phi]`